lib/SPIRV/libSPIRV/SPIRVModule.cpp

```diff
@@ -1772,6 +1772,11 @@ class TopologicalSort {
     ForwardPointerVec.insert(ForwardPointerVec.end(), ForwardPointerSet.begin(),
```

Comment:
Thanks for looking at this leak!

I'm not sure if deleting in `~TopologicalSort` is conceptually the correct fix, because the elements may live longer than `TopologicalSort` it seems.  Perhaps we need to loop over the module's `ForwardPointerVec` instead, although looking at it I'm not sure why the module has that field?

I would like to see what @AlexeySotkin thinks about this.

Reply:
I agree with @svenvh. The problem here is that at line 1718 we allocate a `SPIRVTypeForwardPointer` object but we don't add it to the module. If we add it to the module, it should be deleted in the destructor of `SPIRVTypeForwardPointer` in the loop over `EntryNoId`.

@pauzinl  could you please try to modify line 1718 to something like:
```
SPIRVModule *BM = E->getModule();
ForwardPointerSet.insert(BM->add(new SPIRVTypeForwardPointer(
              BM, Ptr, Ptr->getPointerStorageClass())));
```

Reply:
Changed it in [5e79b32](https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/1092/commits/5e79b3209443f6ddeb049ee62197ceb62b81bc1a)